Replicator/Replicator.cc

```diff
@@ -598,7 +598,9 @@ namespace litecore { namespace repl {
 
         std::for_each(_subRepls.begin(), _subRepls.end(),
                       [](SubReplicator& sub) {
-            sub.checkpointer->stopAutosave();
+            if (sub.checkpointer) {
```

Comment:
Well I found a situation where it is empty, that's why I changed it.  If prepareWorkers throws an error (like if a collection doesn't exist for the passive side) then the checkpointer never gets created.

Reply:
already throw, 
                if (c == nullptr) {
                    error::_throw(error::NotFound, "collection %s is not found in the database.",
                                  _options->collectionOpts[i].collectionPath.asString().c_str());
                }
I guess because it threw, it did not proceed to the ensuing for loop, 
for (CollectionIndex i = 0; i < _options->workingCollectionCount(); ++i) {

Reply:
Perhaps a fix closer to the root-cause could be, before error::throw above, in Replicator::prepareWorkers(),
if (c == nullptr) {
  _subRepls.resize(0);
  error::_throw(...)
}

Reply:
Won't this miss the call to stopAutoSave on any already processed entries?

Reply:
No, we haven't done anything yet. throw will take us out before we do anything

Reply:
Are you going to do this?

Reply:
I swear I did this and now it's gone, but yes....